lib/Services/RetrieveMetadata.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * ownCloud - mediametadata
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Jalpreet Singh Nanda (:imjalpreet) <jalpreetnanda@gmail.com>
+ * @copyright Jalpreet Singh Nanda (:imjalpreet) 2016
+ */
+
+namespace OCA\MediaMetadata\Services;
+
+
+use OCP\IDBConnection;
+
+class RetrieveMetadata {
+
+	protected $dbConnection;
+
+	/**
+	 * @param IDBConnection $connection
+	 */
+	public function __contruct(IDBConnection $connection){
+		$this->dbConnection = $connection;
+	}
+
+	/**
+	 * @param $fileList : List of File IDs
+	 * @return array
+	 */
+	public function retrieve($fileList) {
+
+		$retrievedData = array();
+		$mapper = new ImageDimensionMapper($this->dbConnection);
+
+		$logger = \OC::$server->getLogger();
+
+		foreach ($fileList as $fileID) {
+			$entity = $mapper->find($fileID);
```

Comment:
You don't test `mapper->find()`, you use `setMapperResult` to mock the call to the DB and to return a row containing the values you have defined.

You will assert that the data retrieved from your call to `retrieve` matches the metadata array matching the fileId array.
Because your method uses the mapper, you need to mock those calls.

Let me know if you think of a better way of course.